internal/handlers/common/metadata.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package common provides common code for all handlers.
+package common
+
+import (
+	"context"
+
+	"github.com/FerretDB/FerretDB/internal/clientconn/conninfo"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+)
+
+// CheckClientMetadata checks if the message does not contain client metadata.
+func CheckClientMetadata(ctx context.Context, doc *types.Document) error {
+	if !doc.Has("client") {
+		return nil
+	}
+
+	clientMetadata, err := doc.Get("client")
+	if err != nil {
+		return lazyerrors.Error(err)
+	}
+
+	connInfo := conninfo.Get(ctx)
```

Comment:
Tests explode because `connInfo` is not closed

Reply:
I have tried to add  `defer connInfo.Close()` in:
1.  where you pointed
2. only `TestCheckClientMetadata`, under creating the new context
3. disabling the unit test (`TestCheckClientMetadata`) and keeping 1 point

All scenarios still do explode tests.
I was looking for use cases of closing the connInfo. However, it is unclear to me why.

Reply:
We should call `defer connInfo.Close()` only in handlers code, never in tests.

This place is the right one

Reply:
#3376

Reply:
Unfortunately, it explodes again

Reply:
That's a different panic now, see my other comment